mesonbuild/dependencies/base.py

```diff
@@ -693,10 +693,15 @@ def _call_pkgbin(self, args, env=None):
         sysroot = self.env.properties[self.for_machine].get_sys_root()
         if sysroot:
             env['PKG_CONFIG_SYSROOT_DIR'] = sysroot
-        new_pkg_config_path = ':'.join([p for p in extra_paths])
+        new_pkg_config_path = os.pathsep.join([p for p in extra_paths])
         mlog.debug('PKG_CONFIG_PATH: ' + new_pkg_config_path)
         env['PKG_CONFIG_PATH'] = new_pkg_config_path
 
+        pkg_config_libdir_prop = self.env.properties[self.for_machine].get_pkg_config_libdir()
+        if pkg_config_libdir_prop:
+            new_pkg_config_libdir = os.pathsep.join([p for p in pkg_config_libdir_prop])
```

Comment:
again, this should be `os.pathsep.join(pkg_config_libdir_prop)`

This also isn't safe without ensuring that pkg_config_libdir is not a string, otherwise if it is you'll end up with something like `/:u:s:r:/:l:i:b...`. You're not testing this, but I'm pretty sure it's broken for plain strings as-is.

Reply:
You can use mesonlib.stringlistify(), you can pass it a string or an array of string, and it always return an array of string.

Reply:
sorry i dont understand what you want me to do here.

Reply:
There is nothing to be done here, better do the change in get_pkg_config_libdir().